pkg/odo/cli/delete/component.go

```diff
@@ -119,17 +121,33 @@ func (o *ComponentOptions) deleteDevfileComponent() error {
 	appName := "app"
 
 	log.Info("Searching resources to delete, please wait...")
-	isInnerLoopDeployed, resources, err := o.clientset.DeleteClient.ListResourcesToDeleteFromDevfile(devfileObj, appName)
+	isInnerLoopDeployed, devfileResources, err := o.clientset.DeleteClient.ListResourcesToDeleteFromDevfile(devfileObj, appName)
 	if err != nil {
 		return err
 	}
-	if len(resources) == 0 {
+	if len(devfileResources) == 0 {
 		log.Infof("No resource found for component %q in namespace %q\n", componentName, namespace)
 		return nil
 	}
-
+	var remainingResources []unstructured.Unstructured
+	k8sResources, _ := o.clientset.DeleteClient.ListResourcesToDelete(componentName, namespace)
+	// get resources present in k8sResources(present on the cluster) but not in devfileResources(not present in the devfile)
+	if len(k8sResources) != 0 {
+		for _, k8sresource := range k8sResources {
+			var present bool
+			for _, dresource := range devfileResources {
+				//                                              skip the component's endpoints resource
```

Comment:
```suggestion
				// skip the component's endpoints resource
```

Reply:
Also, there's more happening below than just skipping an endpoint resource. Can you summarize that briefly in the code comment?